mesonbuild/interpreterbase/decorators.py

```diff
@@ -557,6 +561,9 @@ def emit_feature_change(values: T.Dict[_T, T.Union[str, T.Tuple[str, str]]], fea
                         if msg is not None:
                             raise InvalidArguments(f'{name} keyword argument "{info.name}" {msg}')
 
+                    if info.feature_validator is not None:
+                        info.feature_validator(value, subproject, node)
```

Comment:
... and then use `info.feature_validator(value).use(subproject, node)`.

Reply:
It would actually have to be:
```python
if info.feature_validator is not none:
    for feat in info.feature_validator(value):
        feat.use(subproject, node)
```
Which seems like not much of an improvement.